### PR TITLE
Users/bgavril/fixes

### PR DESCRIPTION
**Pull request by bgavrilMS**
Several small changes - see each commit for a description.

### CODE REVIEW

File: Tasks/Maven/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -42,13 +42,19 @@
...
   "loc.input.help.sqDbDetailsRequired": "If using a SonarQube server 5.1 or lower, you must specify the database connection details.",
   "loc.input.label.sqDbUrl": "Db Connection String",
   "loc.input.help.sqDbUrl": "SonarQube server 5.1 and lower only. Enter the database connection setting (i.e. sonar.jdbc.url). For example: jdbc:jtds:sqlserver://localhost/sonar;SelectMethod=Cursor",
   "loc.input.label.sqDbUsername": "Db Username",
   "loc.input.help.sqDbUsername": "SonarQube server 5.1 and lower only. Enter the username for the database user (i.e. sonar.jdbc.username).",
   "loc.input.label.sqDbPassword": "Db User Password",
-  "loc.input.help.sqDbPassword": "SonarQube server 5.1 and lower only. Enter the password for the database user i.e. sonar.jdbc.password"
+  "loc.input.help.sqDbPassword": "SonarQube server 5.1 and lower only. Enter the password for the database user i.e. sonar.jdbc.password",
+  "loc.input.label.pmdAnalysisEnabled": "Run PMD Analysis",
+  "loc.input.help.pmdAnalysisEnabled": "Use the PMD Java static analysis tool to look for bugs in the code.",
```
Comment (bgavrilMS) [inline Tasks/Maven/Strings/resources.resjson/en-US/resources.resjson]: The previous commit omitted the resjson which needs to be checked in as well.

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 [hunk 2]
```diff
@@ -137,30 +137,26 @@ function RetryUntilTrue
...
 
+#
+# Invoke a SonarQube GET Rest API and fetch the response. This method uses the configured SonarQube server host url, as well as basic auth 
+# based on the user provided username / password 
+#
 function InvokeGetRestMethod
 {
-    param (
-        [Parameter(Mandatory=$true)][string]$query, 
-        [bool]$useAuth=$false)
```
Comment (bgavrilMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: All web requests should use auth.

File: Tasks/Maven/task.json [hunk 3]
```diff
@@ -236,7 +236,7 @@
             "type": "connectedService:Generic",
             "label": "SonarQube Endpoint",
             "required": true,
-            "helpMarkDown": "The endpoint that specifies the SonarQube server to use",
+            "helpMarkDown": "The SonarQube server generic endpoint",
```
Comment (bgavrilMS) [inline Tasks/Maven/task.json]: I had a user asking what type of endpoint she should create. So I'm making this clearer.

File: Tests/L0/Common-SonarQubeHelpers/InvokeRestApi.ps1 [hunk 4]
```diff
@@ -41,31 +41,4 @@ Assert-WasCalled Invoke-RestMethod -ParametersEvaluator {
                         $Headers['Authorization'] -eq 'Basic dXNlcjpwYSQkd29yZA==' #this is base64 slang for 'user:pa$$word'
                         }
 
-Unregister-Mock Invoke-RestMethod
-
-
-# Test case 3: authenticated users can make unauthenicated calls to the SQ server
```
Comment (bgavrilMS) [inline Tests/L0/Common-SonarQubeHelpers/InvokeRestApi.ps1]: This test is no longer actual as all calls are now authenticated.

**Comment by blairmc-msft** [comment]
LGTM

File: Tasks/Maven/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (HSAR) [inline Tasks/Maven/Strings/resources.resjson/en-US/resources.resjson]: Ah - thanks very much. This must have been lost in the middle of all the jumping between branches last week.

**Pull request merged**